distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestSingleSignOn.java

```diff
@@ -87,182 +113,252 @@ public void beforeTest() throws Exception {
-        Configuration serverConfig = getAdminConfig()
-                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
-        clientConfig.update(clientSettings);
-        serverConfig.update(serverSettings);
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(clientConfig::getProperties, notNullValue());
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(serverConfig::getProperties, notNullValue());
+        // Get all of the metadata
+        String metadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String ddfSpMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // Make sure all the metadata is valid before we set it
+        validateSaml(metadata, SamlSchema.METADATA);
+        validateSaml(ddfSpMetadata, SamlSchema.METADATA);
+
+        // Set the metadata
+        setConfig("org.codice.ddf.security.idp.client.IdpMetadata", "metadata", metadata);
+        setConfig("org.codice.ddf.security.idp.server.IdpEndpoint", "spMetadata", new String[] {ddfSpMetadata});
     }
 
-    private String getRedirectUrl(Response response) {
-        String fullUrl = null;
+    private void validateSaml(String xml, SamlSchema schema) throws IOException {
 
-        // We can either get a legit redirect from the header, or we can have javascript redirect us.
-        // Whenever javascript redirects us, we have to do ugly, fragile parsing of the HTML.
-        if (response.headers().hasHeaderWithName("Location")) {
-            fullUrl = response.header("Location");
-        } else if (response.body().asString()
-                .contains("<title>Redirect</title>")) { // Javascript redirect
-            Pattern pattern = Pattern.compile("encoded *= *\"(.*)\"", Pattern.CASE_INSENSITIVE);
-            Matcher matcher = pattern.matcher(response.body().asString());
-            matcher.find();
-            fullUrl = matcher.group(1);
-        } else {
-            fail("Unable to extract the redirect URL from the HTTP response. "
-                    + "No redirect found in header and body does not match "
-                    + "regular expression [encoded *= *\"(.*)\"]");
+        // Prepare the schema and xml
+        String schemaFileName = "saml-schema-" + schema.toString().toLowerCase() + "-2.0.xsd";
+        URL schemaURL = getClass().getClassLoader().getResource(schemaFileName);
+        StreamSource streamSource = new StreamSource(new StringReader(xml));
+
+        // If we fail to create a validator we don't want to stop the show, so we just log a warning
+        Validator validator = null;
+        try {
+            validator = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI).newSchema(schemaURL).newValidator();
+        } catch (SAXException e) {
+            LOGGER.warn("Exception creating validator", e.getLocalizedMessage());
```

Comment:
I think here we would rather see the full "e.toString()" here (which includes getLocalizedMessage()), consider changing to:
LOGGER.warn("Exception creating validator. ", e);